League of Memories: New changes report

Deudly opened the pull request:
* Added the necessary functions for passives to update
* Changing the way Recall is handeled

Note that [new `Recall.cs` file can be found in our Spells repository ](https://github.com/LeagueofMemories/LeagueOfMemories-Default/blob/master/Champions/Global/Recall.cs). Unfortunatly, this PR cannot update your submodule since we are not using it

andreivoinea committed:
Changing the way Recall is handeled

I moved the way the recall function works, now it receives a timer (for reduced recall timers and more if we want to) and it handles everything inside the Champion class, so anytime it moves the recall is canceled.

Commit:
Added the necessary functions for passives to update,

Added the four main needed file changes for passives (such as Miss Fortune's Strut) to properly load.

Deudly committed:
Merge pull request #2 from Pokemonred200/indev

Added the necessary functions for passives to update,

Deudly committed:
Fix: A private variable cannot be accessed

Fix: A private variable cannot be accessed

Deudly committed:
Merge pull request #4 from LeagueSandbox/indev

Update: LeagueSandbox/GameServer

Commit:
Added Support for Active Items to LoM. (#5)

Made the changes that enable active items, such as Youmuu's Ghostblade, to function. Additionally, added a fix to Projectile.cs that would have caused issues with Ranged basic attacks.

Commit:
Added new events for ApiEventManager, support for Grievous Wounds, and a class for Utility Helper functions (#6)

With the addition of OnAutoAttackHit, abilities that work with on-hit effects such as Miss Fortune's Impure Shots should now be properly handled. Additionally, support for Grievous Wounds means that abilities that reduce healing effects ,may now function fully. Finally, Helpers.cs will be used for various utility functions, such as Rotate, that are useful for abilities such as Twisted Fate and Graves's Q abilities, which fire in cones.

FurkanS1821 commented on GameServerLib/Logic/GameObjects/AttackableUnits/AI/Champion.cs:
```diff
@@ -67,6 +67,12 @@ public class Champion : ObjAIBase
+
+            for(short i = 0; i < 6; i++)
+            {
+                Spells[(byte)(i + 6)] = new Spell(this, "", (byte)(i + 6));
```
There's exactly what you would want to use as an empty spell: `BaseSpell`.
Riot uses these for non-champion objects, and I believe going that way is better than using an empty string.

FurkanS1821 commented on GameServerLib/Logic/GameObjects/AttackableUnits/AI/Champion.cs:
```diff
@@ -211,7 +225,15 @@ public Vector2 GetRespawnPosition()
+            else
+            {
+                ApiFunctionManager.LogInfo("There is no spell in slot " + slot + ". Using empty spell in its place.");
+                return new Spell(this, "", slot);
```
There's exactly what you would want to use as an empty spell: BaseSpell.
Riot uses these for non-champion objects, and I believe going that way is better than using an empty string.

FurkanS1821 commented on GameServerLib/Logic/GameObjects/AttackableUnits/AI/Champion.cs:
```diff
@@ -245,14 +267,15 @@ public Spell LevelUpSpell(short slot)
-            if (this.isMovementUpdated()) {
-                if (isRecalling)
+			
+			if (this.isMovementUpdated()) {
```
indentation

FurkanS1821 replied:
\+ bracket should be one line below

FurkanS1821 commented on GameServerLib/Logic/GameObjects/AttackableUnits/AI/Champion.cs:
```diff
@@ -329,18 +354,20 @@ public void Respawn()
-            isRecalling = true;
-            visualBuff = ApiFunctionManager.AddBuffHUDVisual("Recall", 8.0f, 1, this);
-            addParticle = ApiFunctionManager.AddParticleTarget(this, "TeleportHome.troy", this);
+			var spawnPos = GetRespawnPosition();
```
indentation

FurkanS1821 commented on GameServerLib/Logic/GameObjects/Spells/Spell.cs:
```diff
@@ -69,7 +69,10 @@ public Spell(Champion owner, string spellName, byte slot)
 
             //Set the game script for the spell
-            spellGameScript = _scriptEngine.CreateObject<GameScript>("Spells", spellName);
+            if (spellName != "")
```
`!spellName.Equals("")`, or even better: `!string.IsNullOrEmpty(spellName)`

here is why: https://stackoverflow.com/a/814881

Deudly committed:
Updated .gitmodules

HarmfulBreeze commented:
Also see my comments on #563

Commit:
Updated files to meet coding standards. (#7)

* Updated files to meet coding standards set by LeagueSandbox's GameServer.

Updated files to meet the change requests made before it's okay to merge with LeagueSandbox's GameServer.

* Fixed Recall up a bit and added a Level Up event.

Recall is now stopped upon damage being taken (horray!) and Red Team can recall due to a function change. Additionally, a level-up event has been added, for champion passives such as Tristana's Draw a Bead.

Commit:
Added model to LaneTurret constructor so turret stats can be loaded (#13)

Fixes the shoot animation

godofloli committed:
Adding EventOnSpellHit and EventOnMoveSuccess (#11)

Adding EventOnSpellHit and EventOnMoveSuccess

Deudly committed:
Revert "Adding EventOnSpellHit and EventOnMoveSuccess (#11)" (#14)

This reverts commit 9b9bbf5e5f119900cc3368f3d66f24e9c0e34cf7.

Deudly committed:
Auto-close GameServer

Auto-close GameServer

Deudly committed:
Revert "Revert "Adding EventOnSpellHit and EventOnMoveSuccess By NurMarvin"" (#15)

Deudly committed:
change

Deudly committed:
Merge branch 'indev' of https://github.com/LeagueofMemories/GameServer into indev

Deudly committed:
added submodules

Deudly committed:
commit

Deudly committed:
change

Deudly committed:
update

Commit:
Fixed navgrid calculations (#16)

- TranslationMaxGridPos wasnt accounting the negative MinGridPos
- MapWidth and MapHeight wasnt calculated correctly, i still think the MinGridPos should be subtracted but hey it works

Litarvan committed:
Fixed Kata spell cast, updated OnDamageTaken, OnDealDamage & OnHitUnit events (#17)

* Added OnUnitDamageDealt event call, updated content submodule

* New events names, deleted useless events

* Fixed mana cost when the champ had exact amount of mana

HarmfulBreeze reviewed (changes requested):
Fix conflicts and the issues Furkan mentioned.
Also please give actual names to your commits, not just "commit" or "update"
Besides that, not much else to say

Commit:
Updated Things to merge with Replication.

Commit:
Merge branch 'Replication-Merger' into indev

Fixed Merge Conflicts.

Commit:
Fixed a few files I accidently broke.

See Above.

Commit:
Removed Helpers.cs.

With the change to Extensions.cs no longer being an internal class, Helpers.cs is no longer necessary.

Commit:
Removed reference to Helpers.cs

HarmfulBreeze commented on .travis.yml:
```diff
@@ -2,13 +2,9 @@ sudo: true
-    - echo "deb http://download.mono-project.com/repo/ubuntu trusty main" | sudo tee /etc/apt/sources.list.d/mono-official.list
-    - sudo apt-get update
-    - sudo apt-get -y install mono-devel
-    - sudo apt-get install nuget
```
Are you sure it still works without installing the nuget package?

HarmfulBreeze reviewed (commented):
Also too many changes! God!

Deudly commented:
Closing this PR since it's useless
